core/src/main/java/org/mule/api/NonBlockingSupported.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api;
+
+import org.mule.api.processor.MessageProcessor;
+
+/**
+ * Marker interface used to mark the {@link org.mule.api.processor.InterceptingMessageProcessor} and
```

Comment:
"Marker interface used to mark the..." is a bit redundant.
Add a separated paragraph for the "Note:" section